Formula/ghc.rb

```diff
@@ -1,18 +1,13 @@
 class Ghc < Formula
   desc "Glorious Glasgow Haskell Compilation System"
   homepage "https://haskell.org/ghc/"
-  url "https://downloads.haskell.org/~ghc/8.10.7/ghc-8.10.7-src.tar.xz"
-  sha256 "e3eef6229ce9908dfe1ea41436befb0455fefb1932559e860ad4c606b0d03c9d"
-  # We bundle a static GMP so GHC inherits GMP's license
-  license all_of: [
-    "BSD-3-Clause",
-    any_of: ["LGPL-3.0-or-later", "GPL-2.0-or-later"],
-  ]
```

Comment:
Actually this may be more accurate since we do build with `--with-intree-gmp` (GMP 6.2.1: https://gitlab.haskell.org/ghc/gmp-tarballs/-/tree/master/).

Looks like we removed GMP for `ghc@9` back in 9.0 but added it back in 9.2.3, but license wasn't changed.

Not sure if the original code to prevent `gmp` linkage on macOS should have been restored?

Reply:
We also bundle a copy of `libffi` so may need `MIT` license too.
```
@rpath/libffi.dylib
```
```
/home/linuxbrew/.linuxbrew/Cellar/ghc/9.2.4/lib/ghc-9.2.4/bin/../rts/libffi.so.7 (ghc)
```